### DISCUSSION TITLE
Anti-periodic boundary conditions

### DISCUSSION THREAD

**Discussion by JonasVeenstra** [zwicker-group/py-pde#217] [Q&A] [answered]
Hi,
Great package!

I'm looking to implement anti-periodic boundary conditions, i.e. the field needs to be constrained to have an opposite sign on either end of the grid. 
I understand how to make BC's that depend on time/space-dependent functions: is there a way to make the BC field-dependent?
Thanks!

**Comment by david-zwicker** [comment, marked as answer]
This is not yet possible within the package, but it should be possible to implement without too much problems. May I ask where such boundary conditions originate (I have never seen such cases). If you could provide a simple example (preferably 1d) where such boundary conditions apply, I could implement them and use the example for automatic testing.

**Reply by JonasVeenstra** [reply]
Hi David,

Thanks for your swift reply! 
I'm trying to simulate a single kink solution of the Sine-Gordon equation on a Möbius strip. Below I write the equation, and the initial state. Only a kink-antikink solution is stable for periodic boundary conditions but I would like to simulate a single kink.

```py
    def evolution_rate(state, t=0):
        u, v = state
        rhs = state.copy()
        rhs[0] = v
        rhs[1] = u.laplace(bc = "natural") - np.sin(u) 
        return rhs

    def get_initial_state(array_u, array_v,L,N):
        grid = CartesianGrid([[0, L]], [N], periodic=True)
        return FieldCollection([ScalarField(grid, array_u),ScalarField(grid, array_v)])

    def kink_IC(x,xc,w=1):
        x=x-xc
        u = 4*np.arctan(np.exp(x/w)) - np.pi
        v = 0*u
        return u,v


```

Does this help?
Thanks!

**Reply by david-zwicker** [reply]
I just merged #219, which introduced anti-periodic boundary conditions. Simply specify `"anti-periodic"` anywhere where boundary conditions are accepted! Let me know if they do what you expect!

**Reply by JonasVeenstra** [reply]
Thanks for the quick implementation. It works exactly as expected. Many thanks!
